io-utils/src/main/java/uk/gov/justice/raml/io/FileTreeScanner.java

```diff
@@ -31,18 +42,35 @@
      * @return a list of paths to matching files under the specified directory
      */
     public Collection<Path> find(final Path baseDir, final String[] includes, final String[] excludes) throws IOException {
+        if (!shouldSearchOnClasspath(baseDir) && !baseDir.toFile().exists()) {
+            return emptyList();
+        }
 
-        LOGGER.debug(format("Finding files in: %s", baseDir.toString()));
+        Reflections reflections = new Reflections(
+                new ConfigurationBuilder()
+                        .filterInputsBy(filterOf(includes, excludes))
+                        .setUrls(shouldSearchOnClasspath(baseDir)
+                                ? ClasspathHelper.forClassLoader()
+                                : asList(baseDir.toUri().toURL()))
+                        .setScanners(new ResourcesScanner()));
+        Set<String> resources = reflections.getResources(Pattern.compile(".*"));
+        return resources.stream().map(Paths::get).collect(toList());
+    }
 
-        DirectoryScanner scanner = new DirectoryScanner();
-        scanner.setBasedir(baseDir.toFile());
-        scanner.setIncludes(includes);
-        scanner.setExcludes(excludes);
-        scanner.scan();
+    private boolean shouldSearchOnClasspath(final Path baseDir) {
+        return baseDir == null || baseDir.toString().contains(CLASSPATH);
+    }
 
-        return asList(scanner.getIncludedFiles())
-                .stream()
-                .map(Paths::get)
-                .collect(Collectors.toList());
+    private Predicate<String> filterOf(final String[] includes, final String[] excludes) {
+        Predicate<String> includesFilter = notEmpty(includes)
+                ? or(stream(includes).map(i -> new AntPathMatcher(i)).collect(toList()))
+                : new AntPathMatcher(RAML_PATTERN);
+        Predicate<String> excludesFilter = and(stream(excludes).map(e -> not(new AntPathMatcher(e))).collect(toList()));
+        return and(includesFilter, excludesFilter);
```

Comment:
I think we should split up the streams/list for each sub-filter into their own lines to make this clearer